<h1>Name: Jade Carlo Redota</h1>
<h2>BSIT-3A</h2>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv('datesets\\penguins.csv')

In [3]:
df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [5]:
df.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [6]:
df.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [7]:
df['bill_length_mm'] = df['bill_length_mm'].mean()

In [8]:
df['bill_length_mm'] = df['bill_length_mm'].fillna(df['bill_length_mm'].mean())

In [9]:
df.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

df['bill_depth_mm'] = df['bill_depth_mm'].fillna(df['bill_depth_mm'].mean())
df['flipper_length_mm'] = df['flipper_length_mm'].fillna(df['flipper_length_mm'].mean())
df['body_mass_g'] = df['body_mass_g'].fillna(df['body_mass_g'].mean())
df['sex'] = df['sex'].fillna(df['sex'].mode()[0])

df.isnull().sum()

<h2>1.How many total records are in the dataset?</h2>

In [10]:
total_length = len(df)

In [11]:
total_length

344

<h3>Insight: The dataset contains 344 total records which makes it easier to identify the patterns and trends in penguin characteristics</h3>

<h2>2.Is there a difference in average flipper length between male and female penguins?</h2>

In [12]:
df.groupby("sex")["flipper_length_mm"].mean()

sex
female    197.363636
male      204.505952
Name: flipper_length_mm, dtype: float64

<h3>Insight: Males generally have slightly longer flippers, which could affect swimming performance or mating displays.</h3>

<h2>3.Which island hosts the most diverse penguin species?</h2>

In [13]:
df.groupby("island")["species"].nunique()

island
Biscoe       2
Dream        2
Torgersen    1
Name: species, dtype: int64

<h3>Insight: The island with the highest unique species count is the Biscoe and Dream .</h3>

<h2>4.How does the average bill length differ among the three penguin species?</h2>

In [14]:
df.groupby("species")["bill_length_mm"].mean()

species
Adelie       43.92193
Chinstrap    43.92193
Gentoo       43.92193
Name: bill_length_mm, dtype: float64

<h3>Insight: The average bill length among the three penguins species is the same 43.92193.</h3>

<h2>5.Which species has the highest average body mass by sex</h2>

In [15]:
df.groupby(["species", "sex"])["body_mass_g"].mean()

species    sex   
Adelie     female    3368.835616
           male      4043.493151
Chinstrap  female    3527.205882
           male      3938.970588
Gentoo     female    4679.741379
           male      5484.836066
Name: body_mass_g, dtype: float64

<h3>Insight: The species that has the highest average body mass by sex is Gentoo is has a 4679.741379 in female,and 5411.011430 in male  .</h3>

<h2>6.How does the average body mass vary across the recorded years?</h2>

In [16]:
df.groupby("year")["body_mass_g"].mean()

year
2007    4124.541284
2008    4266.666667
2009    4210.294118
Name: body_mass_g, dtype: float64

<h3>Insight: The average body that vary across the years is noticeable upward or downward trend could hint at environmental changes or food availability shifts.</h3>

<h2>7.Which year recorded the highest average flipper length?</h2>

In [17]:
df.groupby("island")["flipper_length_mm"].mean()

island
Biscoe       209.706587
Dream        193.072581
Torgersen    191.196078
Name: flipper_length_mm, dtype: float64

<h3>Insight: The highest average in flipper_length_mm is Torgersen it reflect island-specific diets or foraging habits.</h3>

<h2>8.Is there a correlation between bill length and flipper length?</h2>

In [19]:
bill_flipper_cor = df['bill_length_mm'].corr(df['flipper_length_mm'])

bill_flipper_cor

<h3>Insight:A positive correlation would suggest that penguins with longer bills also tend to have longer flippers.
.</h3>

<h2>9.Is there a correlation between bill length and flipper length?</h2>

In [27]:
billdepth_bodymass_cor = df['bill_depth_mm'].corr(df['body_mass_g'])

In [28]:
billdepth_bodymass_cor

np.float64(-0.47191562118606645)

<h3>Insight:Theres no Correlation and could reveal whether thicker-billed penguins are generally heavier.
.</h3>

<h2>10. Is there a correlation between flipper length and body mass?</h2>

In [31]:
flipper_bodymass_cor = df['flipper_length_mm'].corr(df['body_mass_g'])

flipper_bodymass_cor

<h3>Insight:Theres a Correlation and often in animals, larger appendages come with higher body mass this checks if it’s true for penguins.</h3>

<h2>11. Is there a correlation between bill length and bill depth?</h2>

In [35]:
billlength_billdepth_cor = df['bill_length_mm'].corr(df['bill_depth_mm'])

In [36]:
billlength_billdepth_cor

np.float64(-6.479678578715949e-16)

<h3>Insight: Theres no Correlation and it helps understand if penguins with longer bills also tend to have thicker bills.</h3>

<h2>12. Is there a correlation between year and body mass?</h2>

In [38]:
year_bodymass_cor = df['year'].corr(df['body_mass_g'])

In [39]:
year_bodymass_cor

np.float64(0.04220939154335551)

<h3>Insight: Theres no Correlation and while year is categorical in nature here, a correlation could hint at trends in penguin body mass over time.</h3>